app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt

```diff
@@ -216,44 +218,76 @@ class ProfileEditActivityTest {
     launch<ProfileEditActivity>(
       ProfileEditActivity.createProfileEditActivity(
         context = context,
         profileId = 1
       )
     ).use {
-      onView(withId(R.id.profile_delete_button)).perform(click())
-      onView(withText(R.string.profile_edit_delete_dialog_positive))
-        .inRoot(isDialog())
-        .perform(click())
       testCoroutineDispatchers.runCurrent()
-      if (context.resources.getBoolean(R.bool.isTablet)) {
-        intended(hasComponent(AdministratorControlsActivity::class.java.name))
-      } else {
-        intended(hasComponent(ProfileListActivity::class.java.name))
-      }
+
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_title)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_message)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_positive)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_negative)
     }
   }
 
   @Test
-  fun testProfileEdit_landscape_deleteProfile_checkReturnsProfileListOnTabletAdminControlOnPhone() {
+  fun testProfileEdit_deleteProfileSuccessful_dialogIsDisplayedLandscape() {
     launch<ProfileEditActivity>(
       ProfileEditActivity.createProfileEditActivity(
         context = context,
         profileId = 1
       )
     ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_delete_button)).perform(click())
       onView(isRoot()).perform(orientationLandscape())
-      onView(withId(R.id.profile_delete_button)).perform(scrollTo()).perform(click())
-      onView(withText(R.string.profile_edit_delete_dialog_positive))
-        .inRoot(isDialog())
-        .perform(click())
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_title)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_message)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_positive)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_negative)
+    }
+  }
+
+  @Test
+  fun testProfileEdit_deleteProfileSuccessful_clickCancel() {
+    launch<ProfileEditActivity>(
+      ProfileEditActivity.createProfileEditActivity(
+        context = context,
+        profileId = 1
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_title)
+      verifyTextInDialog(textInDialogId = R.string.profile_edit_delete_dialog_message)
+      onView(withText(R.string.profile_edit_delete_dialog_negative)).perform(click())
+      onView(withId(R.id.profile_delete_button)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testProfileEdit_deleteProfile_checkReturnsToProfileListOnPhoneOrAdminControlOnTablet() {
```

Comment:
```suggestion
  fun testProfileEdit_deleteProfile_returnsToProfileListOnPhoneOrAdminControlOnTablet() {
```

'Check' is redundant in this context since it's part of the expected outcome of the test.

Reply:
I have updated this accordingly